mayastor/src/lvs/lvol.rs

```diff
@@ -223,6 +221,48 @@ impl Lvol {
         }
     }
 
+    // wipe the first MB if unmap is not supported on failure the operation
+    // needs to be repeated
+    pub async fn wipe_super(&self) -> Result<(), Error> {
+        if !unsafe { self.0.as_ref().clear_method == LVS_CLEAR_WITH_UNMAP } {
+            let hdl = Bdev::open(&self.as_bdev(), true)
+                .and_then(|desc| desc.into_handle())
+                .map_err(|e| {
+                    error!(?self, ?e, "failed to wipe lvol");
+                    Error::RepDestroy {
+                        source: Errno::ENXIO,
+                        name: self.name(),
+                    }
+                })?;
+            let buf = hdl.dma_malloc(1 << 12).map_err(|e| {
+                error!(
+                    ?self,
+                    ?e,
+                    "no memory available to allocate zero buffer"
+                );
+                Error::RepDestroy {
+                    source: Errno::ENOMEM,
+                    name: self.name(),
+                }
+            })?;
+            // write zero to the first 8MB which whipes the metadata and the
+            // first 4MB of the data partition
+            let range =
+                std::cmp::min(self.as_bdev().size_in_bytes(), (1 << 20) * 8);
+            debug!(?self, ?range, "zeroing range");
+            for offset in 0 .. (range >> 12) {
```

Comment:
We perform 2048 independent I/O operations here ? ( ((1 << 20) * 8) / (1 << 12) = 2048). Can we increase the size of DMA buffer to something like 0.5MB - 1MB to reduce the number of I/Os or there are some other restrictions which I have overlooked ?

Reply:
There is a max on the buffer size we can submit to the bdev layer (64KB iirc) but it would amount to the same thing really. The most important thing here is that buffer is 4K aligned as that is needed for aio and uring. When submitting 64K IOs we would need to ensure that the size of volume is indeed, 64K aligned as well which is not .. i guess all that obvious?

Reply:
I mean, if we're OK with issuing 2048 independent I/O operations, then it's fine.
I heard that modern disks support writing more than 64K of data at once. :-)

Reply:
The max is 64k (SPDK_BDEV_LARGE_BUF_MAX_SIZE).